lib/git-fastclone.rb

```diff
@@ -253,9 +253,11 @@ def clone(url, rev, src_dir, config)
 
         clone_commands = ['git', 'clone', verbose ? '--verbose' : '--quiet']
         # For sparse checkouts, clone directly from the local mirror and skip the actual checkout process
+        # --shared is included so that the checkout remains fast even if the reference and destination directories
```

Comment:
doc says this is a dangerous option which totally makes sense: https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---shared

just to be suare that we have tested cases like deleting the checkout in the machine? yeah this is def only for CI usage only since we have clear idea on when and how to use these clones

Reply:
I think there's potentially room to support more flags to augment or change the behavior of the sparse checkout, but I figure we should only add those if we've got a good reason